libgadget/timestep.c

```diff
@@ -197,7 +197,7 @@ find_timesteps(inttime_t Ti_Current)
         {
             /* make sure the new step is currently active,
              * so that particles do not miss a step */
-            while(!is_timebin_active(bin, Ti_Current) && bin > binold && bin > 1)
+            while((!is_timebin_active(bin, Ti_Current) && bin > binold && bin > 1) || P[pa].Swallowed)
```

Comment:
You don't need this because you already continue; on line 172. Worse, you are indexing with pa, which is an index into ActiveParticle, not P!